zcash_client_sqlite/src/wallet/init.rs

```diff
@@ -310,11 +533,365 @@ mod tests {
+                    nf BLOB NOT NULL UNIQUE,
+                    is_change INTEGER NOT NULL,
+                    memo BLOB,
+                    spent INTEGER,
+                    FOREIGN KEY (tx) REFERENCES transactions(id_tx),
+                    FOREIGN KEY (account) REFERENCES accounts(account),
+                    FOREIGN KEY (spent) REFERENCES transactions(id_tx),
+                    CONSTRAINT tx_output UNIQUE (tx, output_index)
+                )",
+                NO_PARAMS,
+            )?;
+            wdb.conn.execute(
+                "CREATE TABLE IF NOT EXISTS sapling_witnesses (
+                    id_witness INTEGER PRIMARY KEY,
+                    note INTEGER NOT NULL,
+                    block INTEGER NOT NULL,
+                    witness BLOB NOT NULL,
+                    FOREIGN KEY (note) REFERENCES received_notes(id_note),
+                    FOREIGN KEY (block) REFERENCES blocks(height),
+                    CONSTRAINT witness_height UNIQUE (note, block)
+                )",
+                NO_PARAMS,
+            )?;
+            wdb.conn.execute(
+                "CREATE TABLE IF NOT EXISTS sent_notes (
+                    id_note INTEGER PRIMARY KEY,
+                    tx INTEGER NOT NULL,
+                    output_index INTEGER NOT NULL,
+                    from_account INTEGER NOT NULL,
+                    address TEXT NOT NULL,
+                    value INTEGER NOT NULL,
+                    memo BLOB,
+                    FOREIGN KEY (tx) REFERENCES transactions(id_tx),
+                    FOREIGN KEY (from_account) REFERENCES accounts(account),
+                    CONSTRAINT tx_output UNIQUE (tx, output_index)
+                )",
+                NO_PARAMS,
+            )?;
+
+            let address = encode_payment_address(
+                tests::network().hrp_sapling_payment_address(),
+                &extfvk.default_address().1,
+            );
+            let extfvk = encode_extended_full_viewing_key(
+                tests::network().hrp_sapling_extended_full_viewing_key(),
+                extfvk,
+            );
+            wdb.conn.execute(
+                "INSERT INTO accounts (account, extfvk, address)
+                VALUES (?, ?, ?)",
+                &[
+                    u32::from(account).to_sql()?,
+                    extfvk.to_sql()?,
+                    address.to_sql()?,
+                ],
+            )?;
+
+            Ok(())
+        }
+
+        let seed = [0xab; 32];
+        let account = AccountId::from(0);
+        let secret_key = sapling::spending_key(&seed, tests::network().coin_type(), account);
+        let extfvk = ExtendedFullViewingKey::from(&secret_key);
+        let data_file = NamedTempFile::new().unwrap();
+        let mut db_data = WalletDb::for_path(data_file.path(), tests::network()).unwrap();
+        init_0_3_0(&mut db_data, &extfvk, account).unwrap();
+        init_wallet_db(&mut db_data, Secret::new(seed.to_vec())).unwrap();
```

Comment:
Do we want to do any checks on the resulting db here (as opposed to checking just that `init_wallet_db` succeeds)?

Reply:
If we need additional checks, they should actually be implemented inside the migrations themselves, so that they will abort the migration if our invariants are violated. I've done this with WalletMigration2, and WalletMigration0 and WalletMigration1 don't have any invariants that need to hold.

Reply:
Do you mean the foreign key constraints? It doesn't look as if those will be enforced for the migration, because of the shipping of `PRAGMA foreign_keys = OFF;` as I pointed out at https://github.com/zcash/librustzcash/pull/600/files#r946241489 .

Reply:
Got it. I fixed that scoping of the pragmas now.